mutt/string.c

```diff
@@ -858,3 +858,66 @@ void mutt_str_hyphenate(char *buf, size_t buflen, const char *str)
       *buf = '-';
   }
 }
+
+/**
+ * mutt_str_inbox_cmp - Do two folders share the same path and one is an inbox - @ingroup sort_api
+ * @param a First path
+ * @param b Second path
+ * @retval -1 a is INBOX of b
+ * @retval  0 None is INBOX
+ * @retval  1 b is INBOX for a
+ *
+ * This function compares two folder paths. It first looks for the position of
+ * the last common '/' character. If a valid position is found and it's not the
+ * last character in any of the two paths, the remaining parts of the paths are
+ * compared (case insensitively) with the string "INBOX" followed by a non
+ * alpha character, e.g., '.' or '/'. If only one of the two paths matches,
+ * it's reported as being less than the other and the function returns -1 (a <
+ * b) or 1 (a > b).  If both or no paths match the requirements, the two paths
+ * are considered equivalent and this function returns 0.
+ *
+ * Examples:
+ * * mutt_str_inbox_cmp("/foo/bar",      "/foo/baz") --> 0
+ * * mutt_str_inbox_cmp("/foo/bar/",     "/foo/bar/inbox") --> 0
+ * * mutt_str_inbox_cmp("/foo/bar/sent", "/foo/bar/inbox") --> 1
+ * * mutt_str_inbox_cmp("=INBOX",        "=Drafts") --> -1
+ * * mutt_str_inbox_cmp("=INBOX",        "=INBOX.Foo") --> 0
+ * * mutt_str_inbox_cmp("=INBOX.Foo",    "=Drafts") --> -1 
+ */
+int mutt_str_inbox_cmp(const char *a, const char *b)
+{
+#define IS_INBOX(s) (mutt_istrn_equal(s, "inbox", 5) && !isalnum((s)[5]))
+#define CMP_INBOX(a, b) (IS_INBOX(b) - IS_INBOX(a))
+
+  /* fast-track in case the paths have been mutt_pretty_mailbox'ified */
+  if ((a[0] == '+') && (b[0] == '+'))
+  {
+    return CMP_INBOX(a + 1, b + 1);
+  }
+
+  const char *a_end = strrchr(a, '/');
+  const char *b_end = strrchr(b, '/');
+
+  /* If one path contains a '/', but not the other */
+  if ((!a_end) ^ (!b_end))
+    return 0;
+
+  /* If neither path contains a '/' */
+  if (!a_end)
```

Comment:
The comment doesn't match the code?

Reply:
Well it does, in a way. Just above, I check `(!a_end) ^ (!b_end)`. So here, `!a_end` ↔ `!b_end`.